src/test/scala/loamstream/drm/uger/QacctAccountingClientTest.scala

```diff
@@ -12,6 +13,113 @@ final class QacctAccountingClientTest extends FunSuite {
 
   import QacctTestHelpers.actualQacctOutput
 
+  test("delaySequence") {
+    val delays = QacctAccountingClient.delaySequence
+    
+    val actual = delays.take(10).toIndexedSeq
+    
+    import scala.concurrent.duration._
+    
+    val expected = Seq(
+        0.5.seconds, 
+        1.second, 
+        2.seconds, 
+        4.seconds, 
+        8.seconds, 
+        16.seconds,
+        30.seconds,
+        30.seconds,
+        30.seconds,
+        30.seconds)
+        
+    assert(actual === expected)
+  }
```

Comment:
Just wanted to toss out a possible different assert/test suggestion. Instead of comparing the _exact_ durations expected (and an exact cap), I could see it better if:

* The cap was in code (paired with `delaySequence`) so it's somewhat configurable; and
* It would be possible to adjust the sequence later with ease and not impact the test.

I'm thinking something like...

```
val seqOk = actual.foldLeft(true -> 0.seconds) { 
  case ((ok, a), b) => (ok && b > a || b == delaySeqCap) -> b 
}._1

assert(seqOk)
```

It will ensure that each subsequent delay is greater than the first or that the cap was reached.

Just a thought.

Reply:
Note: the reason I was thinking about this was - even though I haven't pulled and ran tests (yet) - I was thinking that for testing failures, you won't want to wait the full 30+ seconds to test retries, and instead might want to use a different delay sequence, which would make the test that does exist fail.

Reply:
> The cap was in code (paired with delaySequence) so it's somewhat configurable; 
This is a good idea, I'll do that.

> val seqOk = actual.foldLeft(true -> 0.seconds) { 
  case ((ok, a), b) => (ok && (b > a || b == delaySeqCap)) -> b 
}._1

I'm less crazy about this.  I had to think through the fold a bit (a little bit, but still).  In general, I'd rather test that a function produced the actual thing it was supposed to, instead of that what's produced was in line with some more-abstract properties.  A sequence like "-10 seconds, 1 day, a squillion years" would pass the above, but wouldn't be right.  

I think there's a place for both testing with known inputs and expectations, and property-oriented testing, I just feel better with the former here.

> you won't want to wait the full 30+ seconds to test retries
A good point.  Other tests here only do two retries to avoid waiting for too long, but that sounds more like a point in favor of making the start and cap for the sequence configurable.

Reply:
Ya, I didn't like the fold either (hard to reason), but it's what I slapped together in 10 seconds just as an example. :-)